src/main/java/com/nekromant/telegram/service/ResumeAnalysisRequestService.java

```diff
@@ -1,17 +1,91 @@
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 
+import javax.ws.rs.core.MediaType;
+
+
+@Slf4j
 @Service
 public class ResumeAnalysisRequestService {
     @Autowired
     private ResumeAnalysisRequestRepository resumeAnalysisRequestRepository;
+    @Autowired
+    private PaymentDetailsRepository paymentDetailsRepository;
+    @Autowired
+    private UserInfoService userInfoService;
+    @Autowired
+    private MentoringReviewBot mentoringReviewBot;
+    @Autowired
+    private LifePayFeign lifePayFeign;
+    @Autowired
+    private TelegramFeign telegramFeign;
+    @Autowired
+    private CVAnalProp props;
+
 
-    public void save(ResumeAnalysisRequest resumeAnalysisRequest) {
+    public void save(byte[] bytes, String tgName, String phone) {
+        ResumeAnalysisRequest resumeAnalysisRequest = new ResumeAnalysisRequest();
+        resumeAnalysisRequest.setCVPdf(bytes);
+        resumeAnalysisRequest.setTgName(tgName);
+        resumeAnalysisRequest.setPhone(phone);
+
+        String description = "Оплата за разбор резюме по договору публичной оферты";
+        ChequeDTO chequeDTO = new ChequeDTO(props.getLogin(), props.getApikey(), props.getAmount(), description, phone, props.getMethod());
+
+        log.info("Sending request to LifePay" + chequeDTO);
+        LifePayResponseDTO lifePayResponse = new Gson().fromJson(lifePayFeign.payCheque(chequeDTO).getBody(), LifePayResponseDTO.class);
+        log.info("LifePay response: " + lifePayResponse);
+
+        PaymentDetails paymentDetails = new PaymentDetails();
+        paymentDetails.setNumber(lifePayResponse.getData().getNumber());
+        paymentDetails.setStatus(PayStatus.UNREDEEMED.get());
+        paymentDetailsRepository.save(paymentDetails);
+        log.info("Unredeemed payment created: " + paymentDetails);
+
+        resumeAnalysisRequest.setLifePayNumber(lifePayResponse.getData().getNumber());
         resumeAnalysisRequestRepository.save(resumeAnalysisRequest);
+        log.info("New resume analysis request created: " + resumeAnalysisRequest);
     }
 
+    public void sendCVToMentorForAnalysisOrReject(PaymentDetails paymentDetails) {
+        paymentDetailsRepository.deleteByNumber(paymentDetails.getNumber());
+        paymentDetailsRepository.save(paymentDetails);
+
+        if (paymentDetails.getStatus().equals(PayStatus.FAIL.get())) {
+            log.info("Payment failed: " + paymentDetails);
+            resumeAnalysisRequestRepository.deleteByLifePayNumber(paymentDetails.getNumber());
+            return;
+        }
+
+        log.info("Payment details have been redeemed:" + paymentDetails);
+
+        byte[] CV_bytes = resumeAnalysisRequestRepository.findByLifePayNumber(paymentDetails.getNumber()).getCVPdf();
+        FormData formData = new FormData(MediaType.MULTIPART_FORM_DATA, "document", CV_bytes);
+        String receiverId = userInfoService.getUserInfo(props.getReceiverName()).getChatId().toString();
```

Comment:
этот метод вернется userId только если юзер пользовался ботом, иначе не найдется
думаю с апи телеги стоит тянуть айди по имени

Reply:
а вот пишут, что id можно получить только если пользователь пользовался ботом, да и то не всегда.

Unfortunately, there is no direct way to get a Telegram user's ID by their username without interaction with any bot. This is because Telegram does not provide a public API endpoint or method that allows you to look up a user's ID based on their username.
The only way to get a Telegram user's ID by their username is through direct interaction with a bot that has been given permission to access the user's information. When a user interacts with a bot, the bot receives information about the user, including their user ID and username, which can then be used by the bot to identify the user in future interactions.
Without this direct interaction, it is not possible to retrieve a user's ID based on their username alone. This is for privacy and security reasons, as Telegram values user privacy and does not allow arbitrary access to user information without their consent.

Reply:
а это айдишник админа(ментора) и есть) просто так неочевидно его получаю